sdk/src/sharedTest/java/io/customer/sdk/hook/HookManagerTest.kt

```diff
@@ -0,0 +1,62 @@
+package io.customer.sdk.hook
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.customer.common_test.BaseTest
+import io.customer.sdk.hooks.CioHooksManager
+import io.customer.sdk.hooks.HookModule
+import io.customer.sdk.hooks.ModuleHook
+import io.customer.sdk.hooks.ModuleHookProvider
+import io.customer.sdk.utils.random
+import org.amshove.kluent.internal.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class HookManagerTest : BaseTest() {
+
+    private lateinit var cioHooksManager: CioHooksManager
+
+    @Before
+    override fun setup() {
+        super.setup()
+
+        cioHooksManager = CioHooksManager()
+    }
+
+    @Test
+    fun subscribeToUpdate_givenSubscribedToHooksManager_expectGetHookUpdate() {
+
+        val identifier = String.random
+        val profileIdentifiedHook = ModuleHook.ProfileIdentifiedHook(identifier)
+
+        val beforeProfileStoppedBeingIdentifiedHook = ModuleHook.BeforeProfileStoppedBeingIdentified(identifier)
+
+        val screen = String.random
+        val screenTrackedHook = ModuleHook.ScreenTrackedHook(screen)
+
+        cioHooksManager.add(
+            HookModule.MessagingInApp,
+            object : ModuleHookProvider() {
+                override fun profileIdentifiedHook(hook: ModuleHook.ProfileIdentifiedHook) {
+                    assertEquals(hook, profileIdentifiedHook)
+                    assertEquals(hook.identifier, identifier)
+                }
+
+                override fun beforeProfileStoppedBeingIdentified(hook: ModuleHook.BeforeProfileStoppedBeingIdentified) {
+                    assertEquals(hook, beforeProfileStoppedBeingIdentifiedHook)
+                    assertEquals(hook.identifier, identifier)
+                }
+
+                override fun screenTrackedHook(hook: ModuleHook.ScreenTrackedHook) {
+                    assertEquals(hook, screenTrackedHook)
+                    assertEquals(hook.screen, screen)
+                }
+            }
+        )
+
+        cioHooksManager.onHookUpdate(profileIdentifiedHook)
+        cioHooksManager.onHookUpdate(beforeProfileStoppedBeingIdentifiedHook)
+        cioHooksManager.onHookUpdate(screenTrackedHook)
```

Comment:
What is the reason for calling 3 separate hook functions in this test function?

Reply:
Ah, ok. The test function's job is to call each of the module hook event types and make sure that the callbacks are called for each type. 

I agree that this test function provides value. My main concern is the maintainability of it. If we were to add a new hook event type to the project, we would need to remember to go into this test function and add another `cioHooksManager.onHookUpdate()` call with the new hook event type. 

To avoid this issue, I wonder if there is a way to use an `enum` or something similar? Something that allows us to write code like this:

```kotlin
HookEventTypes.forEach { hookEventType -> 
  cioHooksManager.onHookUpdate(hookEventType)
} 
```

The code could be very different then what's above, but the idea is that we write a test function once and then anytime that a new module hook event type is added to the project, the test function will automatically test that new event type?

Reply:
I don't think that would be quite possible, since hook types are expecting parameters, `identify` would require an `identifier` similarly, and others would require their respective parameters and those parameters are validated as well so, can't be sending `any`

> we would need to remember to go into this test function

So what I can do to resolve the remembering part, is make it an enum so, it forces you to handle the newly added hook type.

Reply:
It looks like this has been implemented [here](https://github.com/customerio/customerio-android/blob/ccca81d528cbeeb458290debedaecc27c2b6eff3/sdk/src/sharedTest/java/io/customer/sdk/hook/HookManagerTest.kt#L74-L83=). 

Because the test function is a bit long and *could* be simplified by someone in the future, I suggest adding a comment to the test function explaining why the test function is written this way. A comment saying that the test function is designed to remind the developer to update the test function when a new hook event type is added. 

Besides that, this looks to be resolved.